src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/multiclass_nms/multiclass_nms_kernel_ref.cpp

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "multiclass_nms_kernel_ref.h"
+
+#include <algorithm>
+#include <string>
+
+#include "kernel_selector_utils.h"
+
+namespace kernel_selector {
+
+ParamsKey MulticlassNmsKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableInputDataType(Datatype::F32);
+    k.EnableInputDataType(Datatype::INT32);
+    k.EnableInputDataType(Datatype::INT64);
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+    k.EnableOutputDataType(Datatype::INT32);
+    k.EnableOutputDataType(Datatype::INT64);
+    k.EnableAllInputLayout();
+    k.EnableAllOutputLayout();
+
+    k.EnableBatching();
+    k.EnableDifferentTypes();
+    k.EnableTensorPitches();
+
+    return k;
+}
+
+KernelsPriority MulticlassNmsKernelRef::GetKernelsPriority(const Params&, const optional_params&) const {
+    return DONT_USE_IF_HAVE_SOMETHING_ELSE;
+}
+
+bool MulticlassNmsKernelRef::Validate(const Params& p, const optional_params& o) const {
+    if (p.GetType() != KernelType::MULTICLASS_NMS || o.GetType() != KernelType::MULTICLASS_NMS) {
+        return false;
+    }
+
+    // FIXME opoluektov: more checks on the attribute values
+    return true;
+}
+
+namespace {
+MulticlassNmsKernelRef::DispatchData SetDefault(const multiclass_nms_params& params, size_t idx) {
+    MulticlassNmsKernelRef::DispatchData dispatch_data;
+
+    if (idx == 0 || idx == 2) {
+        const auto num_batches = params.has_roisnum ? params.inputs[2].Batch().v : params.inputs[1].Batch().v;
+        dispatch_data.gws = {num_batches, 1, 1};
+    } else {
+        dispatch_data.gws = {1, 1, 1};
+    }
+    dispatch_data.lws = GetOptimalLocalWorkGroupSizes(dispatch_data.gws, params.engineInfo);
+
+    return dispatch_data;
+}
+}  // namespace
+
+void MulticlassNmsKernelRef::SetKernelArguments(const multiclass_nms_params& params,
+                                                size_t idx, cldnn::arguments_desc& arguments) const {
+    for (auto i = 0; i < 2 + params.has_roisnum + 2; ++i) {
+        arguments.push_back({ArgumentDescriptor::Types::INPUT, (uint32_t)i});
+    }
+    arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+    arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 0});
+}
+
+JitConstants MulticlassNmsKernelRef::GetJitConstants(const multiclass_nms_params& params) const {
+    JitConstants jit = MakeBaseParamsJitConstants(params);
+
+    const auto num_batches = params.has_roisnum ? params.inputs[2].Batch().v : params.inputs[1].Batch().v;
+    const int64_t num_classes = params.has_roisnum ? params.inputs[0].Batch().v : params.inputs[1].Feature().v;
+    const auto num_boxes = params.inputs[0].Feature().v;
+
+    auto real_num_classes = num_classes;
+    if (params.background_class >= 0 && params.background_class < num_classes) {
+        real_num_classes = std::max(static_cast<int64_t>(1), num_classes - 1);
```

Comment:
Nope. This static_cast is introduced to fix Windows build (since now warnings are treated as errors on Windows as well). 1ull won't work even on Linux - params.backgound_classes (and num_classes accordingly) is *signed* .

Reply:
1LL also?

Reply:
Got rid of static_cast - and let's see if it works for Windows build )